lsp-mode.el

```diff
@@ -4690,8 +4690,10 @@ returns the command to execute."
   (list :connect (lambda (filter sentinel name)
                    (let* ((final-command (lsp-resolve-final-function local-command))
                           ;; wrap with stty to disable converting \r to \n
-                          (wrapped-command (append '("stty" "raw" ";") final-command))
-                          (process-name (generate-new-buffer-name name)))
+                          (process-name (generate-new-buffer-name name))
+                          (wrapped-command (append '("stty" "raw" ";")
+                                                   final-command
+                                                   (list (format "2>%s" (make-temp-file process-name))))))
```

Comment:
`make-temp-file` produces a tempfile specific to the Emacs host, but the LSP server runs remotely. Edebug shows the value of `wrapped-command` at this point is:

```
("stty" "raw" ";" "pyls" "2>/var/folders/2w/fgshrnj48x01s7pr7x00010gn/T/pyls-remote4kwB7c")
```

That path (almost certainly) does not exist on the LSP server host, which causes the attempt to start the LSP server to fail on startup.

Reply:
Yep, it will be even more problematic if you do tramp Windows->Linux. Updated PR to use /tmp as a default path and I added an option to specify a function to generate the temp fine on the remote machine.

Reply:
Cool. With your latest update, the code works. I suggested a change to make the stderr redirect files more unique.